app/workers/pubsubhubbub/distribution_worker.rb

```diff
@@ -12,10 +12,10 @@ def perform(stream_entry_id)
 
     account = stream_entry.account
     payload = AtomSerializer.render(AtomSerializer.new.feed(account, [stream_entry]))
-    domains = account.followers_domains
+    domains = account.followers_subscription_domains if stream_entry.status&.hidden?
 
     Subscription.where(account: account).active.select('id, callback_url').find_each do |subscription|
-      next unless domains.include?(Addressable::URI.parse(subscription.callback_url).host)
+      next if stream_entry.status&.hidden? && !domains.include?(Addressable::URI.parse(subscription.callback_url).normalize.host)
```

Comment:
In my last proposal (well, before the last commit which switches to a database field), I am already using “.uniq” to avoid .normalize()-ing URLs from the same domain, as it's the most expensive part of the computation. This does however require the use of a large array at some point, indeed.

Checking it in Pubsubhubbub::DeliveryWorker would involve doing the whole thing for each PuSH subscriber (so, roughly multiply the total running time by the number of remote instances), and avoiding the big array means you will do multiple a .normalize() call for *each* follower, even if multiple ones happen to use the same domain…